python/ml4ir/base/io/spark_io.py

```diff
@@ -0,0 +1,119 @@
+import json
+import yaml
+from pyspark import SparkContext
+from pyspark.sql import SparkSession
+
+HDFS_PREFIX = "hdfs://"
+
+
+class SparkConfigHolder:
+    """Class to store configuration objects for Spark to avoid duplication"""
+
+    HADOOP_CONFIG = None
+    HDFS = None
+    LOCAL_FS = None
+
+
+def get_spark_session() -> SparkSession:
+    """Get or create Spark Session"""
+    return SparkSession.builder.appName("ml4ir").getOrCreate()
+
+
+def get_spark_context() -> SparkContext:
+    """Get spark context"""
+    return get_spark_session().sparkContext
+
+
+def get_path_from_str(file_path):
+    """Convert a string file path to org.apache.hadoop.fs.Path object"""
+    return get_spark_context()._gateway.jvm.org.apache.hadoop.fs.Path(file_path)
+
+
+def get_hadoop_config():
+    """Get hadoop configuration object"""
+    if not SparkConfigHolder.HADOOP_CONFIG:
+        SparkConfigHolder.HADOOP_CONFIG = get_spark_context()._jsc.hadoopConfiguration()
+
+    return SparkConfigHolder.HADOOP_CONFIG
+
+
+def get_hdfs():
```

Comment:
@balikasg This is the main module to review. I had to jump some hoops to get access to the hadoop File system since it is only available in spark's java based APIs.

Reply:
thanks for the pointer!
From what I understand it is not a Flowsnake v2 thing, we are expecting HDFS paths and we are handling things as this.